csharp/ql/lib/semmlecode.csharp.dbscheme

```diff
@@ -1316,6 +1311,12 @@ lambda_expr_return_type(
   unique int id: @lambda_expr ref,
   int type_id: @type_or_ref ref);
 
+/* Compiler generated */
+
+@generatable = @expr | @stmt | @modifiable;
+
+compiler_generated(unique int id: @generatable ref);
```

Comment:
I think this could be `@element` to cover the general case. I could argue that the `using` directives added by the `ImplicitUsings` setting are also compiler generated. Similarly, `TargetFrameworkAttribute`s in `AssemblyAttributes.cs` are also compiler generated.

Reply:
Ok, then we just do `compiler_generated(unique int id: @element ref);`. Then it is also easier to add more compiler generated stuff later without changing the DB Scheme again.
I will make the change.